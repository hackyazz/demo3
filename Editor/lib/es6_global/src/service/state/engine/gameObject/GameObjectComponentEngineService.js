

import * as GameObjectAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/GameObjectAPI.js";
import * as GetComponentGameObjectService$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/gameObject/GetComponentGameObjectService.js";

function getLightMaterialComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getLightMaterialComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function getBasicMaterialComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getBasicMaterialComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function getMeshRendererComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getMeshRendererComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

var unsafeGetGeometryComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectGeometryComponent;

function getGeometryComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getGeometryComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function hasLightComponent(gameObject, engineState) {
  if (GameObjectAPI$Wonderjs.hasGameObjectDirectionLightComponent(gameObject, engineState)) {
    return true;
  } else {
    return GameObjectAPI$Wonderjs.hasGameObjectPointLightComponent(gameObject, engineState);
  }
}

var hasLightMaterialComponent = GameObjectAPI$Wonderjs.hasGameObjectLightMaterialComponent;

var unsafeGetLightMaterialComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectLightMaterialComponent;

var addLightMaterialComponent = GameObjectAPI$Wonderjs.addGameObjectLightMaterialComponent;

var disposeLightMaterialComponent = GameObjectAPI$Wonderjs.disposeGameObjectLightMaterialComponent;

var removeLightMaterialComponent = GameObjectAPI$Wonderjs.removeGameObjectLightMaterialComponent;

var hasBasicMaterialComponent = GameObjectAPI$Wonderjs.hasGameObjectBasicMaterialComponent;

var unsafeGetBasicMaterialComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectBasicMaterialComponent;

var addBasicMaterialComponent = GameObjectAPI$Wonderjs.addGameObjectBasicMaterialComponent;

var disposeBasicMaterialComponent = GameObjectAPI$Wonderjs.disposeGameObjectBasicMaterialComponent;

var removeBasicMaterialComponent = GameObjectAPI$Wonderjs.removeGameObjectBasicMaterialComponent;

var addMeshRendererComponent = GameObjectAPI$Wonderjs.addGameObjectMeshRendererComponent;

var unsafeGetMeshRendererComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectMeshRendererComponent;

var hasMeshRendererComponent = GameObjectAPI$Wonderjs.hasGameObjectMeshRendererComponent;

var disposeMeshRendererComponent = GameObjectAPI$Wonderjs.disposeGameObjectMeshRendererComponent;

var disposeGeometryComponent = GameObjectAPI$Wonderjs.disposeGameObjectGeometryComponent;

var removeGeometryComponent = GameObjectAPI$Wonderjs.removeGameObjectGeometryComponent;

var hasGeometryComponent = GameObjectAPI$Wonderjs.hasGameObjectGeometryComponent;

var addGeometryComponent = GameObjectAPI$Wonderjs.addGameObjectGeometryComponent;

var addPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.addGameObjectPerspectiveCameraProjectionComponent;

var hasPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.hasGameObjectPerspectiveCameraProjectionComponent;

var unsafeGetPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectPerspectiveCameraProjectionComponent;

var disposePerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.disposeGameObjectPerspectiveCameraProjectionComponent;

var addBasicCameraViewComponent = GameObjectAPI$Wonderjs.addGameObjectBasicCameraViewComponent;

var hasBasicCameraViewComponent = GameObjectAPI$Wonderjs.hasGameObjectBasicCameraViewComponent;

var unsafeGetBasicCameraViewComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectBasicCameraViewComponent;

var disposeBasicCameraViewComponent = GameObjectAPI$Wonderjs.disposeGameObjectBasicCameraViewComponent;

var unsafeGetTransformComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectTransformComponent;

var hasTransformComponent = GameObjectAPI$Wonderjs.hasGameObjectTransformComponent;

var unsafeGetSourceInstanceComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectSourceInstanceComponent;

var hasSourceInstanceComponent = GameObjectAPI$Wonderjs.hasGameObjectSourceInstanceComponent;

var addSourceInstanceComponent = GameObjectAPI$Wonderjs.addGameObjectSourceInstanceComponent;

var addDirectionLightComponent = GameObjectAPI$Wonderjs.addGameObjectDirectionLightComponent;

var disposeDirectionLightComponent = GameObjectAPI$Wonderjs.disposeGameObjectDirectionLightComponent;

var unsafeGetDirectionLightComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectDirectionLightComponent;

var hasDirectionLightComponent = GameObjectAPI$Wonderjs.hasGameObjectDirectionLightComponent;

var addPointLightComponent = GameObjectAPI$Wonderjs.addGameObjectPointLightComponent;

var disposePointLightComponent = GameObjectAPI$Wonderjs.disposeGameObjectPointLightComponent;

var unsafeGetPointLightComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectPointLightComponent;

var hasPointLightComponent = GameObjectAPI$Wonderjs.hasGameObjectPointLightComponent;

var addArcballCameraControllerComponent = GameObjectAPI$Wonderjs.addGameObjectArcballCameraControllerComponent;

var disposeArcballCameraControllerComponent = GameObjectAPI$Wonderjs.disposeGameObjectArcballCameraControllerComponent;

var unsafeGetArcballCameraControllerComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectArcballCameraControllerComponent;

var hasArcballCameraControllerComponent = GameObjectAPI$Wonderjs.hasGameObjectArcballCameraControllerComponent;

var getAllLightMaterialComponents = GameObjectAPI$Wonderjs.getAllLightMaterialComponents;

var getAllDirectionLightComponents = GameObjectAPI$Wonderjs.getAllDirectionLightComponents;

var getAllBasicCameraViewComponents = GameObjectAPI$Wonderjs.getAllBasicCameraViewComponents;

var getAllArcballCameraControllerComponents = GameObjectAPI$Wonderjs.getAllArcballCameraControllerComponents;

var getAllPerspectiveCameraProjectionComponents = GameObjectAPI$Wonderjs.getAllPerspectiveCameraProjectionComponents;

var getAllGeometryComponents = GameObjectAPI$Wonderjs.getAllGeometryComponents;

export {
  hasLightMaterialComponent ,
  getLightMaterialComponent ,
  unsafeGetLightMaterialComponent ,
  addLightMaterialComponent ,
  disposeLightMaterialComponent ,
  removeLightMaterialComponent ,
  hasBasicMaterialComponent ,
  getBasicMaterialComponent ,
  unsafeGetBasicMaterialComponent ,
  addBasicMaterialComponent ,
  disposeBasicMaterialComponent ,
  removeBasicMaterialComponent ,
  addMeshRendererComponent ,
  getMeshRendererComponent ,
  unsafeGetMeshRendererComponent ,
  hasMeshRendererComponent ,
  disposeMeshRendererComponent ,
  unsafeGetGeometryComponent ,
  getGeometryComponent ,
  disposeGeometryComponent ,
  removeGeometryComponent ,
  hasGeometryComponent ,
  addGeometryComponent ,
  addPerspectiveCameraProjectionComponent ,
  hasPerspectiveCameraProjectionComponent ,
  unsafeGetPerspectiveCameraProjectionComponent ,
  disposePerspectiveCameraProjectionComponent ,
  addBasicCameraViewComponent ,
  hasBasicCameraViewComponent ,
  unsafeGetBasicCameraViewComponent ,
  disposeBasicCameraViewComponent ,
  unsafeGetTransformComponent ,
  hasTransformComponent ,
  unsafeGetSourceInstanceComponent ,
  hasSourceInstanceComponent ,
  addSourceInstanceComponent ,
  addDirectionLightComponent ,
  disposeDirectionLightComponent ,
  unsafeGetDirectionLightComponent ,
  hasDirectionLightComponent ,
  addPointLightComponent ,
  disposePointLightComponent ,
  unsafeGetPointLightComponent ,
  hasPointLightComponent ,
  hasLightComponent ,
  addArcballCameraControllerComponent ,
  disposeArcballCameraControllerComponent ,
  unsafeGetArcballCameraControllerComponent ,
  hasArcballCameraControllerComponent ,
  getAllLightMaterialComponents ,
  getAllDirectionLightComponents ,
  getAllBasicCameraViewComponents ,
  getAllArcballCameraControllerComponents ,
  getAllPerspectiveCameraProjectionComponents ,
  getAllGeometryComponents ,
  
}
/* GameObjectAPI-Wonderjs Not a pure module */
