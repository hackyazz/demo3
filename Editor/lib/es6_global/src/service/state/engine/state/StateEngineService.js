

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
