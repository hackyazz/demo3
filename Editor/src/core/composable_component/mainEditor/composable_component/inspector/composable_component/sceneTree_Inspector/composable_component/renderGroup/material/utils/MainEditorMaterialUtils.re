open SelectType;

open MainEditorMaterialType;

open MaterialDataAssetType;

let getMaterialOptions = () => [|
  {key: BasicMaterial |> convertMaterialTypeToInt, value: "basic_material"},
  {key: LightMaterial |> convertMaterialTypeToInt, value: "light_material"},
|];

let getMaterialTypeByGameObject = (gameObject, engineState) =>
  switch (
    GameObjectComponentEngineService.hasBasicMaterialComponent(
      gameObject,
      engineState,
    ),
    GameObjectComponentEngineService.hasLightMaterialComponent(
      gameObject,
      engineState,
    ),
  ) {
  | (true, false) => BasicMaterial
  | (false, true) => LightMaterial
  | _ =>
    WonderLog.Log.fatal(
      LogUtils.buildFatalMessage(
        ~description=
          {j|gameObject:$gameObject should has material component|j},
        ~reason="",
        ~solution={j||j},
        ~params={j||j},
      ),
    )
  };

let handleSpecificFuncByMaterialType =
    (type_, (handleBasicMaterialFunc, handleLightMaterialFunc)) =>
  switch (type_) {
  | BasicMaterial => handleBasicMaterialFunc()

  | LightMaterial => handleLightMaterialFunc()
  };

let createMaterialByType = (type_, engineState) =>
  switch (type_) {
  | BasicMaterial => BasicMaterialEngineService.create(engineState)

  | LightMaterial => LightMaterialEngineService.create(engineState)
  };

let getMaterialComponentByType = (gameObject, type_, engineState) =>
  switch (type_) {
  | BasicMaterial =>
    GameObjectComponentEngineService.unsafeGetBasicMaterialComponent(
      gameObject,
      engineState,
    )
  | LightMaterial =>
    GameObjectComponentEngineService.unsafeGetLightMaterialComponent(
      gameObject,
      engineState,
    )
  };

let renameMaterialByMaterialType =
    (newName, type_, materialComponent, engineState) =>
  switch (type_) {
  | BasicMaterial =>
    BasicMaterialEngineService.setBasicMaterialName(
      materialComponent,
      newName,
      engineState,
    )

  | LightMaterial =>
    LightMaterialEngineService.setLightMaterialName(
      materialComponent,
      newName,
      engineState,
    )
  };

let getGameObjectsByType = (material, type_, engineState) =>
  switch (type_) {
  | BasicMaterial =>
    BasicMaterialEngineService.getBasicMaterialGameObjects(
      material,
      engineState,
    )
  | LightMaterial =>
    LightMaterialEngineService.getLightMaterialGameObjects(
      material,
      engineState,
    )
  };

let unsafeGetGameObjectsByType = (material, type_, engineState) =>
  getGameObjectsByType(material, type_, engineState)
  |> OptionService.unsafeGet;