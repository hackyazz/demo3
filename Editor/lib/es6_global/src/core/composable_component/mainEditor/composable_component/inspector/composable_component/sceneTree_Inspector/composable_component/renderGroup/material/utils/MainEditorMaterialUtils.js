

import * as Curry from "../../../../../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Log$WonderLog from "../../../../../../../../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as LogUtils$WonderEditor from "../../../../../../../../../../utils/console/LogUtils.js";
import * as OptionService$WonderEditor from "../../../../../../../../../../../service/primitive/OptionService.js";
import * as BasicMaterialEngineService$WonderEditor from "../../../../../../../../../../../service/state/engine/BasicMaterialEngineService.js";
import * as LightMaterialEngineService$WonderEditor from "../../../../../../../../../../../service/state/engine/LightMaterialEngineService.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../../../../../../service/state/engine/gameObject/GameObjectComponentEngineService.js";

function getMaterialOptions(param) {
  return /* array */[
          /* record */[
            /* key : BasicMaterial */0,
            /* value */"basic_material"
          ],
          /* record */[
            /* key : LightMaterial */1,
            /* value */"light_material"
          ]
        ];
}

function getMaterialTypeByGameObject(gameObject, engineState) {
  var match = GameObjectComponentEngineService$WonderEditor.hasBasicMaterialComponent(gameObject, engineState);
  var match$1 = GameObjectComponentEngineService$WonderEditor.hasLightMaterialComponent(gameObject, engineState);
  var exit = 0;
  if (match) {
    if (match$1) {
      exit = 1;
    } else {
      return /* BasicMaterial */0;
    }
  } else if (match$1) {
    return /* LightMaterial */1;
  } else {
    exit = 1;
  }
  if (exit === 1) {
    return Log$WonderLog.fatal(LogUtils$WonderEditor.buildFatalMessage("gameObject:" + (String(gameObject) + " should has material component"), "", "", ""));
  }
  
}

function handleSpecificFuncByMaterialType(type_, param) {
  if (type_) {
    return Curry._1(param[1], /* () */0);
  } else {
    return Curry._1(param[0], /* () */0);
  }
}

function createMaterialByType(type_, engineState) {
  if (type_) {
    return LightMaterialEngineService$WonderEditor.create(engineState);
  } else {
    return BasicMaterialEngineService$WonderEditor.create(engineState);
  }
}

function getMaterialComponentByType(gameObject, type_, engineState) {
  if (type_) {
    return GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(gameObject, engineState);
  } else {
    return GameObjectComponentEngineService$WonderEditor.unsafeGetBasicMaterialComponent(gameObject, engineState);
  }
}

function renameMaterialByMaterialType(newName, type_, materialComponent, engineState) {
  if (type_) {
    return LightMaterialEngineService$WonderEditor.setLightMaterialName(materialComponent, newName, engineState);
  } else {
    return BasicMaterialEngineService$WonderEditor.setBasicMaterialName(materialComponent, newName, engineState);
  }
}

function getGameObjectsByType(material, type_, engineState) {
  if (type_) {
    return LightMaterialEngineService$WonderEditor.getLightMaterialGameObjects(material, engineState);
  } else {
    return BasicMaterialEngineService$WonderEditor.getBasicMaterialGameObjects(material, engineState);
  }
}

function unsafeGetGameObjectsByType(material, type_, engineState) {
  return OptionService$WonderEditor.unsafeGet(getGameObjectsByType(material, type_, engineState));
}

export {
  getMaterialOptions ,
  getMaterialTypeByGameObject ,
  handleSpecificFuncByMaterialType ,
  createMaterialByType ,
  getMaterialComponentByType ,
  renameMaterialByMaterialType ,
  getGameObjectsByType ,
  unsafeGetGameObjectsByType ,
  
}
/* Log-WonderLog Not a pure module */
