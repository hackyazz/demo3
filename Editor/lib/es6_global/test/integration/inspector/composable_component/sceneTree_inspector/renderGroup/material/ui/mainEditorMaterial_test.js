

import * as Curry from "../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ReactTestTool$WonderEditor from "../../../../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../../../tool/GameObjectTool.js";
import * as EventListenerTool$WonderEditor from "../../../../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../../../tool/BuildComponentTool.js";
import * as DirectorToolEngine$WonderEditor from "../../../../../../../tool/engine/DirectorToolEngine.js";
import * as StateEngineService$WonderEditor from "../../../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../../../tool/MainEditorSceneTool.js";
import * as MainEditorAssetIdTool$WonderEditor from "../../../../../../asset/tool/MainEditorAssetIdTool.js";
import * as AssetTreeInspectorTool$WonderEditor from "../../../../assetTree_inspector/tool/AssetTreeInspectorTool.js";
import * as MeshRendererToolEngine$WonderEditor from "../../../../../../../tool/engine/MeshRendererToolEngine.js";
import * as LightMaterialToolEngine$WonderEditor from "../../../../../../../tool/engine/LightMaterialToolEngine.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../../../../../asset/tool/MainEditorAssetTreeTool.js";
import * as MainEditorMaterialUtils$WonderEditor from "../../../../../../../../src/core/composable_component/mainEditor/composable_component/inspector/composable_component/sceneTree_Inspector/composable_component/renderGroup/material/utils/MainEditorMaterialUtils.js";
import * as MainEditorBasicMaterialTool$WonderEditor from "../tool/MainEditorBasicMaterialTool.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../../../../../../tool/engine/NoWorkerJobConfigToolEngine.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../../../src/service/state/engine/GameObjectComponentEngineService.js";
import * as MainEditorAssetHeaderOperateNodeTool$WonderEditor from "../../../../../../asset/tool/MainEditorAssetHeaderOperateNodeTool.js";

describe("MainEditorMaterial", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, undefined, undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test set currentSceneTreeNode", (function () {
                beforeEach((function () {
                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                      }));
                describe("test change material", (function () {
                        describe("test snapshot", (function () {
                                return Wonder_jest.test("test show default light material component", (function () {
                                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMaterial(undefined, undefined, undefined, undefined, /* () */0));
                                            }));
                              }));
                        describe("test logic", (function () {
                                beforeEach((function () {
                                        return DirectorToolEngine$WonderEditor.prepareAndInitAllEnginState(/* () */0);
                                      }));
                                Wonder_jest.test("currentSceneTreeNode's default material should be lightMaterial", (function () {
                                        var partial_arg = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                        var materialType = StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                return MainEditorMaterialUtils$WonderEditor.getMaterialTypeByGameObject(partial_arg, param);
                                              }));
                                        return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](materialType), /* LightMaterial */1);
                                      }));
                                describe("test change currentSceneTreeNode's lightMaterial to basicMaterial", (function () {
                                        Wonder_jest.test("test currentSceneTreeNode's material component should be basicMaterial", (function () {
                                                MainEditorBasicMaterialTool$WonderEditor.changeMaterialTypeToBeBasicMaterial(undefined, undefined, /* () */0);
                                                var partial_arg = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                                return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                      return GameObjectComponentEngineService$WonderEditor.hasBasicMaterialComponent(partial_arg, param);
                                                                    }))), true);
                                              }));
                                        Wonder_jest.test("test gameObject should move from lightMaterialRenderArray to basicMaterialRenderArray", (function () {
                                                var match = MeshRendererToolEngine$WonderEditor.getAllRenderArrayCount(/* () */0);
                                                MainEditorBasicMaterialTool$WonderEditor.changeMaterialTypeToBeBasicMaterial(undefined, undefined, /* () */0);
                                                return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](MeshRendererToolEngine$WonderEditor.getAllRenderArrayCount(/* () */0)), /* tuple */[
                                                            match[0] + 2 | 0,
                                                            match[1] - 2 | 0
                                                          ]);
                                              }));
                                        return Wonder_jest.test("should remove lightMaterial instead of dispose it", (function () {
                                                      var currentGameObject = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                                      var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                                      var oldLightMaterial = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(currentGameObject, engineState);
                                                      MainEditorBasicMaterialTool$WonderEditor.changeMaterialTypeToBeBasicMaterial(undefined, undefined, /* () */0);
                                                      var engineState$1 = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                                      LightMaterialToolEngine$WonderEditor.isAlive(oldLightMaterial, engineState$1),
                                                                      GameObjectComponentEngineService$WonderEditor.hasLightMaterialComponent(currentGameObject, engineState$1)
                                                                    ]), /* tuple */[
                                                                  true,
                                                                  false
                                                                ]);
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        return /* () */0;
                      }));
                describe("deal with specific case", (function () {
                        describe("test MainEditorMaterialUtils getMaterialTypeByGameObject function", (function () {
                                return Wonder_jest.test("test if gameObject haven't material component, should throw error", (function () {
                                              return Wonder_jest.Expect[/* toThrowMessageRe */21]((/should\shas\smaterial\scomponent/g), Wonder_jest.Expect[/* expect */0]((function () {
                                                                MainEditorSceneTool$WonderEditor.setDirectionLightGameObjectToBeCurrentSceneTreeNode(/* () */0);
                                                                var partial_arg = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                                                return StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                              return MainEditorMaterialUtils$WonderEditor.getMaterialTypeByGameObject(partial_arg, param);
                                                                            }));
                                                              })));
                                            }));
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        describe("test select material group->show order", (function () {
                beforeEach((function () {
                        MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                        MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                        return /* () */0;
                      }));
                return Wonder_jest.test("\n        order should be:\n        1)default material is in the end;\n        2)sort material assets by firstname alphabetically\n        ", (function () {
                              var addedMaterialNodeId1 = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                              var addedMaterialNodeId2 = addedMaterialNodeId1 + 1 | 0;
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                              AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetMaterialNode */2](undefined, undefined, addedMaterialNodeId1, "BMaterial", /* () */0);
                              AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetMaterialNode */2](undefined, undefined, addedMaterialNodeId2, "AMaterial", /* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMaterial(GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0), undefined, undefined, true, /* () */0));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
