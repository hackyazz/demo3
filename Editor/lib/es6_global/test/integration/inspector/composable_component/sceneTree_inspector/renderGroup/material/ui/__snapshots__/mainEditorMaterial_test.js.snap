// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorMaterial test select material group->show order 
        order should be:
        1)default material is in the end;
        2)sort material assets by firstname alphabetically
         1`] = `
<article
  className="wonder-inspector-material"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Material
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Material
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          AMaterial
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          BMaterial
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Basic-Material
        </div>
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
  <div
    className="material-value"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        <span
          className=""
        >
          Type
        </span>
      </div>
      <div
        className="item-content"
      >
        <select
          onChange={[Function]}
          value="1"
        >
          <option
            value="0"
          >
            basic_material
          </option>
          <option
            value="1"
          >
            light_material
          </option>
        </select>
      </div>
    </article>
    <div
      className=""
    >
      <article
        className="wonder-light-material"
      >
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffcuse color
          </div>
          <div
            className="item-content item-color"
          >
            <div
              className="color-hex"
              onClick={[Function]}
              style={
                Object {
                  "background": "#ffffff",
                }
              }
            />
            <div
              className="color-select"
              onClick={[Function]}
            >
              <img
                src="./public/img/color.png"
              />
            </div>
          </div>
        </article>
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffuse map
          </div>
          <div
            className="item-content item-texture"
          >
            <div
              className="texture-img"
              onClick={[Function]}
              style={
                Object {
                  "opacity": "1",
                }
              }
            >
              <div
                className="img-dragBg"
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
              />
            </div>
            <button
              className="texture-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </article>
        <article
          className="inspector-item wonder-float-input"
        >
          <div
            className="item-header component-label"
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            Shininess
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </article>
      </article>
    </div>
    <div
      className="material-shade"
    />
  </div>
</article>
`;

exports[`MainEditorMaterial test set currentSceneTreeNode test change material test snapshot test show default light material component 1`] = `
<article
  className="wonder-inspector-material"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Material
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="material-value"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        <span
          className=""
        >
          Type
        </span>
      </div>
      <div
        className="item-content"
      >
        <select
          onChange={[Function]}
          value="1"
        >
          <option
            value="0"
          >
            basic_material
          </option>
          <option
            value="1"
          >
            light_material
          </option>
        </select>
      </div>
    </article>
    <div
      className=""
    >
      <article
        className="wonder-light-material"
      >
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffcuse color
          </div>
          <div
            className="item-content item-color"
          >
            <div
              className="color-hex"
              onClick={[Function]}
              style={
                Object {
                  "background": "#ffffff",
                }
              }
            />
            <div
              className="color-select"
              onClick={[Function]}
            >
              <img
                src="./public/img/color.png"
              />
            </div>
          </div>
        </article>
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffuse map
          </div>
          <div
            className="item-content item-texture"
          >
            <div
              className="texture-img"
              onClick={[Function]}
              style={
                Object {
                  "opacity": "1",
                }
              }
            >
              <div
                className="img-dragBg"
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
              />
            </div>
            <button
              className="texture-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </article>
        <article
          className="inspector-item wonder-float-input"
        >
          <div
            className="item-header component-label"
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            Shininess
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </article>
      </article>
    </div>
    <div
      className="material-shade"
    />
  </div>
</article>
`;
